db/migrate/20240806061243_create_accounts_table.rb

```diff
@@ -0,0 +1,14 @@
+class CreateAccountsTable < ActiveRecord::Migration[7.1]
+  def change
+    create_table :accounts, id: false do |t|
+      t.string :account_number, primary_key: true
+      t.string :pin
+      t.decimal :balance, precision: 10, scale: 2
+      t.string :status
+      t.string :currency
```

Comment:
for your purposes, you can store integer and create a mapping for it in the model. it saves database space.